tests/kafkatest/tests/streams/streams_broker_compatibility_test.py

```diff
@@ -55,20 +56,28 @@ def setUp(self):
         self.zk.start()
 
    
-    @parametrize(broker_version=str(LATEST_0_10_2))
-    @parametrize(broker_version=str(LATEST_0_10_1))
-    def test_fail_fast_on_incompatible_brokers_if_eos_enabled(self, broker_version):
```

Comment:
This is covered via `test_fail_fast_on_incompatible_brokers` that tests against 0.10.0, 0.10.1 and 0.10.2 already

In older KS versions, 0.10.1 and 0.10.2 where only not supported if EOS is used, but now those older brokers are not supported at all any longer, and thus, we can merge both tests.